// modules/footer.js
'use strict';

import React from 'react';

import Actions from 'flux/actions';
import StudioContact from 'components/studio-contact';

const studios = [{
  id: 'london',
  street: '62 Shoreditch High Street',
  city: 'London',
  region: null,
  postcode: 'E1 6JJ',
  country: 'United Kingdom',
  map: 'https://maps.google.com/maps?q=62+Shoreditch+High+Street+London+E1+6JJ+United+Kingdom&hl=en&sll=40.747244,-73.995747&sspn=0.014972,0.023432&gl=us&hnear=62+Shoreditch+High+St,+London,+United+Kingdom&t=m&z=16'
}, {
  id: 'newyork',
  street: '26 Broadway 16th Floor',
  city: 'New York',
  region: 'NY.',
  postcode: '10004',
  country: 'United States',
  map: 'https://www.google.com/maps/place/26+Broadway,+New+York,+NY+10004,+USA/@40.7053319,-74.0129945,17z/data=!3m1!4b1!4m2!3m1!1s0x89c25a115d64bc3d:0xd453ec88640e1e02'
}, {
  id: 'malmo',
  street: 'Södra Förstadsgatan 2',
  city: 'Malmö',
  region: null,
  postcode: '211 43',
  country: 'Sweden',
  map: 'https://maps.google.com/maps?q=S%C3%B6dra+F%C3%B6rstadsgatan+2+211+43+Malm%C3%B6+Sweden&hl=en&ie=UTF8&sll=51.524115,-0.077254&sspn=0.012296,0.023432&gl=us&hnear=S%C3%B6dra+F%C3%B6rstadsgatan+2,+211+43+Malm%C3%B6,+Sweden&t=m&z=16'
}, {
  id: 'sydney',
  street: '199 Cleveland St',
  city: 'Sydney',
  region: 'NSW',
  postcode: '2016',
  country: 'Australia',
  map: 'https://www.google.co.uk/maps/place/199+Cleveland+St,+Redfern+NSW+2016,+Australia/@-33.88968,151.203396,17z/data=!3m1!4b1!4m2!3m1!1s0x6b12b1d8db4b96d9:0x7f17818603b1c5a0?hl=en'
}];

export default class Navigation extends React.Component {
  renderStudio(studio) {
    return <StudioContact studio={studio} />;
  }
  onClickShowContacts(e) {
    e.preventDefault();
    Actions.toggleContacts();
  }
  render() {
    const facebookLogo = '<use xlink:href="images/spritemap.svg#facebook" />';
    const twitterLogo = '<use xlink:href="images/spritemap.svg#twitter" />';
    const linkedinLogo = '<use xlink:href="images/spritemap.svg#linkedin" />';
    return (
      <footer className="footer">
        <div className="content">
          <div className="general">
            <h1 className="contact-title">Contact</h1>
            <hr className="contact-divider" />
            <a className="email-cta" href="mailto:studio@ustwo.com" onClick={this.onClickShowContacts}>Get in touch</a>
            <ul className="social">
              <li className="channel facebook">
                <a href="http://www.facebook.com/ustwo">
                  <svg className="logo" title="facebook logo" role="img" dangerouslySetInnerHTML={{__html: facebookLogo }} />
                </a>
              </li>
              <li className="channel twitter">
                <a href="http://www.twitter.com/ustwo">
                  <svg className="logo" title="twitter logo" role="img" dangerouslySetInnerHTML={{__html: twitterLogo }} />
                </a>
              </li>
              <li className="channel linkedin">
                <a href="http://www.linkedin.com/ustwo">
                  <svg className="logo" title="linkedin logo" role="img" dangerouslySetInnerHTML={{__html: linkedinLogo }} />
                </a>
              </li>
            </ul>
          </div>
          <ul className="studios">
            {studios.map(this.renderStudio)}
          </ul>
          <p className="copyright">Copyright &copy; 2015 ustwo studio Ltd. All rights reserved.</p>
        </div>
      </footer>
    );
  }
};
