// modules/footer.js
'use strict';

import React from 'react';

import StudioContact from 'components/studio-contact';

const studios = [{
  id: 'london',
  street: '62 Shoreditch High Street',
  city: 'London',
  region: null,
  postcode: 'E1 6JJ',
  country: 'United Kingdom',
  map: 'https://www.google.co.uk/maps/place/ustwo+London/@51.524064,-0.077065,17z/data=!4m6!1m3!3m2!1s0x48761cb0c877dff1:0x4b063d67310deeb0!2sustwo+London!3m1!1s0x48761cb0c877dff1:0x4b063d67310deeb0'
}, {
  id: 'newyork',
  street: '26 Broadway 16th Floor',
  city: 'New York',
  region: 'NY.',
  postcode: '10004',
  country: 'United States',
  map: 'https://www.google.co.uk/maps/place/ustwo+London/@51.524064,-0.077065,17z/data=!4m6!1m3!3m2!1s0x48761cb0c877dff1:0x4b063d67310deeb0!2sustwo+London!3m1!1s0x48761cb0c877dff1:0x4b063d67310deeb0'
}, {
  id: 'malmo',
  street: 'Södra Förstadsgatan 2',
  city: 'Malmö',
  region: null,
  postcode: '211 43',
  country: 'Sweden',
  map: 'https://www.google.co.uk/maps/place/ustwo+London/@51.524064,-0.077065,17z/data=!4m6!1m3!3m2!1s0x48761cb0c877dff1:0x4b063d67310deeb0!2sustwo+London!3m1!1s0x48761cb0c877dff1:0x4b063d67310deeb0'
}, {
  id: 'sydney',
  street: '199 Cleveland St',
  city: 'Sydney',
  region: 'NSW',
  postcode: '2016',
  country: 'Australia',
  map: 'https://www.google.co.uk/maps/place/ustwo+London/@51.524064,-0.077065,17z/data=!4m6!1m3!3m2!1s0x48761cb0c877dff1:0x4b063d67310deeb0!2sustwo+London!3m1!1s0x48761cb0c877dff1:0x4b063d67310deeb0'
}];

export default class Navigation extends React.Component {
  renderStudio(studio) {
    return <StudioContact studio={studio} />;
  }
  render() {
    const facebookLogo = '<use xlink:href="images/spritemap.svg#facebook" />';
    const twitterLogo = '<use xlink:href="images/spritemap.svg#twitter" />';
    const linkedinLogo = '<use xlink:href="images/spritemap.svg#linkedin" />';
    return (
      <footer className="footer">
        <div className="content">
          <div className="general">
            <h1 className="contact-title">Contact</h1>
            <hr className="contact-divider" />
            <a className="email-cta" href="mailto:studio@ustwo.com">Get in touch</a>
            <ul className="social">
              <li className="channel facebook">
                <a href="http://www.facebook.com/ustwo">
                  <svg className="logo" title="facebook logo" role="img" dangerouslySetInnerHTML={{__html: facebookLogo }} />
                </a>
              </li>
              <li className="channel twitter">
                <a href="http://www.twitter.com/ustwo">
                  <svg className="logo" title="twitter logo" role="img" dangerouslySetInnerHTML={{__html: twitterLogo }} />
                </a>
              </li>
              <li className="channel linkedin">
                <a href="http://www.linkedin.com/ustwo">
                  <svg className="logo" title="linkedin logo" role="img" dangerouslySetInnerHTML={{__html: linkedinLogo }} />
                </a>
              </li>
            </ul>
          </div>
          <ul className="studios">
            {studios.map(this.renderStudio)}
          </ul>
        </div>
      </footer>
    );
  }
};
