// templates/page-home.js
'use strict';

import React from 'react';
import Imager from 'imager.jsx';
import ScrollMagic from 'ScrollMagic';
import 'ScrollMagic-Indicators';

import DownChevron from 'elements/down-chevron';
import WordAnimation from 'elements/word-animation';
import BoldHeader from 'components/bold-header';
import HomeTextBlock from 'components/home-text-block';
import ScreenBlock from 'components/screen-block';
import Rotator from 'elements/rotator';
import EntranceTransition from 'elements/entrance-transition';

export default class PageHome extends React.Component {
  scrollState = {
    top: false,
    middle: false,
    bottom: false
  }
  constructor(props) {
    super(props);
    this.state = {
      blocks: [
        {
          blockReference: () => {return React.findDOMNode(this.refs.blockWelcome)},
          hexColour: '6114CC'
        },
        {
          blockReference: () => {return React.findDOMNode(this.refs.blockClient)},
          hexColour: '6A86EC'
        },
        {
          blockReference: () => {return React.findDOMNode(this.refs.blockOwnStuff)},
          hexColour: 'FFBF02'
        },
        {
          blockReference: () => {return React.findDOMNode(this.refs.blockVenture)},
          hexColour: 'F9615B'
        }
      ]
    };
  }
  animateChevron = (event) => {
    this.refs.downChevron.resetAnim();
    this.refs.downChevron.anim();
  }
  scrollDownPage = (event) => {
    TweenMax.to(React.findDOMNode(this).parentElement, 0.8, {scrollTo:{y:this.state.blocks[0].blockReference().clientHeight}, ease:Power2.easeOut});
  }
  setupScrollMagic = () => {
    this.scrollController = new ScrollMagic.Controller();

    let blockWelcome = this.state.blocks[0].blockReference();
    let scrollSceneWelcome = new ScrollMagic.Scene({
        triggerElement: blockWelcome,
        triggerHook: 'onLeave',
        duration: () => {return blockWelcome.clientHeight * 0.7}
      })
      .addTo(this.scrollController)
      .on('progress', (e) => {
        if (e.progress > 0) {
          this.refs.downChevron.goToProgressRatio(0.91 - e.progress * 0.9);
        }
    });

    let pageReference = React.findDOMNode(this);
    // set initial colour – we need to do this due to having an offset
    pageReference.style.backgroundColor = '#' + this.state.blocks[0].hexColour;
    this.state.blocks.forEach((block, index) => {
      block.blockReference().style.backgroundColor = 'transparent';
      if (index > 0) {
        this.createColourBlockScene(pageReference, block.blockReference(), this.state.blocks[index - 1].hexColour, block.hexColour);
      }
    });

    let pageHeight = pageReference.clientHeight;
    let foldHeight = blockWelcome.clientHeight;
    let scrollHeightRatio = (pageHeight - foldHeight) / pageHeight;
    this.scrollSceneTracker = new ScrollMagic.Scene({
        triggerElement: pageReference,
        triggerHook: 'onLeave',
        duration: () => {return pageReference.clientHeight}
      })
      .addTo(this.scrollController)
      .on('progress', (e) => {
        if(!this.scrollState.top && e.progress >= foldHeight / pageHeight * scrollHeightRatio) {
          this.scrollState.top = true;
          ga('send', 'event', 'hub_page', 'scroll_1', 'home');
        }
        if(!this.scrollState.middle && e.progress > scrollHeightRatio / 2) {
          this.scrollState.middle = true;
          ga('send', 'event', 'hub_page', 'scroll_2', 'home');
        }
        if(!this.scrollState.bottom && e.progress > scrollHeightRatio) {
          this.scrollState.bottom = true;
          ga('send', 'event', 'hub_page', 'scroll_3', 'home');
        }
    });
  }
  createColourBlockScene = (pageReference, blockReference, hexColour1, hexColour2) => {
    return new ScrollMagic.Scene({
        triggerElement: blockReference,
        triggerHook: 'onEnter',
        offset: blockReference.clientHeight * 0.25,
        duration: () => {return blockReference.clientHeight * 0.5}
      })
      .addTo(this.scrollController)
      // .addIndicators() // add indicators (requires plugin)
      .on('progress', (e) => {
        window.requestAnimationFrame(() => {
          pageReference.style.backgroundColor = '#' + this.blendColours(hexColour1, hexColour2, e.progress);
        });
    });
  }
  blendColours = (colour1, colour2, percentage) => {
    function intToHex(num) {
      let hex = Math.round(num).toString(16);
      return hex.length === 1 ? '0' + hex : hex;
    }

    function hexToInt(hexArray) {
      return [parseInt(hexArray[0] + hexArray[1], 16), parseInt(hexArray[2] + hexArray[3], 16), parseInt(hexArray[4] + hexArray[5], 16)];
    }

    let rgbColour1 = hexToInt(colour1);
    let rgbColour2 = hexToInt(colour2);

    let rgbColour3 = [
      (1 - percentage) * rgbColour1[0] + percentage * rgbColour2[0],
      (1 - percentage) * rgbColour1[1] + percentage * rgbColour2[1],
      (1 - percentage) * rgbColour1[2] + percentage * rgbColour2[2]
    ];

    return intToHex(rgbColour3[0]) + intToHex(rgbColour3[1]) + intToHex(rgbColour3[2]);
  }
  componentDidMount() {
    if(!env.modernizr.touchevents) {
      this.setupScrollMagic();
    }
    this.animTimeout = setTimeout(() => {
      this.animateChevron();
    }, 2500);
  }
  componentWillUnmount() {
    this.scrollController.destroy(true);
    clearTimeout(this.animTimeout);
  }
  render() {
    const HeadlineBackground = Imager({});
    const HarveyNicksDevice = Imager({});
    const HarveyNicksShape1 = '<use xlink:href="images/spritemap.svg#HarveyShape1" />';
    const HarveyNicksShape2 = '<use xlink:href="images/spritemap.svg#HarveyShape2" />';
    const HarveyNicksShape3 = '<use xlink:href="images/spritemap.svg#HarveyShape3" />';
    const HarveyNicksShape4 = '<use xlink:href="images/spritemap.svg#HarveyShape4" />';
    const MonumentDevice = Imager({});
    const MonumentAward1 = '<use xlink:href="images/spritemap.svg#MonumentAward1" />';
    const MonumentAward2 = '<use xlink:href="images/spritemap.svg#MonumentAward2" />';
    const MonumentAward3 = '<use xlink:href="images/spritemap.svg#MonumentAward3" />';
    const DiceScreenAnim = Imager({});
    const DiceDevice = Imager({});
    const headlineWordsAnimationOptions = {
      ease: Power2.easeOut,
      opacity: 0,
      y: 30
    };
    return (
      <article className="page-home">
        <ScreenBlock ref="blockWelcome" customClass="page-home__screen-block--welcome" hexColour={'#' + this.state.blocks[0].hexColour}>
          <EntranceTransition className="entrance">
            <HeadlineBackground className="page-home__screen-block--welcome__headline-background-image" src="images/home/Homepage_hero_00.png"/>
            <Rotator delay={1400} duration={2000} interval={4000} keep={3}>
              <HeadlineBackground key="image1" className="page-home__screen-block--welcome__headline-background-image" src="images/home/Homepage_hero_01.png"/>
              <HeadlineBackground key="image2" className="page-home__screen-block--welcome__headline-background-image" src="images/home/Homepage_hero_02.png"/>
              <HeadlineBackground key="image3" className="page-home__screen-block--welcome__headline-background-image" src="images/home/Homepage_hero_03.png"/>
              <HeadlineBackground key="image4" className="page-home__screen-block--welcome__headline-background-image" src="images/home/Homepage_hero_04.png"/>
              <HeadlineBackground key="image5" className="page-home__screen-block--welcome__headline-background-image" src="images/home/Homepage_hero_05.png"/>
              <HeadlineBackground key="image6" className="page-home__screen-block--welcome__headline-background-image" src="images/home/Homepage_hero_06.png"/>
            </Rotator>
          </EntranceTransition>
          <div className="page-home__screen-block--welcome__header-vertical-centerer">
            <BoldHeader customClass="page-home__screen-block--welcome__header-vertical-centerer__bold-header" colour="white">
              <WordAnimation delay={1} duration={0.4} options={headlineWordsAnimationOptions}>
                We're a digital product studio
              </WordAnimation>
            </BoldHeader>
          </div>
          <DownChevron customClass="page-home__screen-block--welcome__down-chevron" ref="downChevron" onClick={this.scrollDownPage}/>
        </ScreenBlock>
        <ScreenBlock ref="blockClient" customClass="page-home__screen-block--client" hexColour={'#' + this.state.blocks[1].hexColour}>
          <div className="page-home__screen-block--client__image-vertical-centerer-parent">
            <div className="page-home__screen-block--client__image-vertical-centerer-parent__image-vertical-centerer-child">
              <div className="page-home__screen-block--client__image-vertical-centerer-parent__image-vertical-centerer-child__image-margin-container">
                <svg className="page-home__screen-block--client__image-vertical-centerer-parent__image-vertical-centerer-child__image-margin-container__harvey-nicks-shape-3" role="presentation" dangerouslySetInnerHTML={{__html: HarveyNicksShape3 }} />
                <svg className="page-home__screen-block--client__image-vertical-centerer-parent__image-vertical-centerer-child__image-margin-container__harvey-nicks-shape-4" role="presentation" dangerouslySetInnerHTML={{__html: HarveyNicksShape4 }} />
                <HarveyNicksDevice className="page-home__screen-block--client__image-vertical-centerer-parent__image-vertical-centerer-child__image-margin-container__harvey-nicks-device" src="images/home/Homepage_Harvey_Phone.png"/>
                <svg className="page-home__screen-block--client__image-vertical-centerer-parent__image-vertical-centerer-child__image-margin-container__harvey-nicks-shape-1" role="presentation" dangerouslySetInnerHTML={{__html: HarveyNicksShape1 }} />
                <svg className="page-home__screen-block--client__image-vertical-centerer-parent__image-vertical-centerer-child__image-margin-container__harvey-nicks-shape-2" role="presentation" dangerouslySetInnerHTML={{__html: HarveyNicksShape2 }} />
              </div>
            </div>
          </div>
          <div className="page-home__screen-block--client__text-block-vertical-centerer">
            <HomeTextBlock title="Innovative client work">
              <p>We work as partners with the world’s best businesses to help them build successful, innovative digital products for a complex and evolving marketplace.</p>
            </HomeTextBlock>
          </div>
        </ScreenBlock>
        <ScreenBlock ref="blockOwnStuff" customClass="page-home__screen-block--own-stuff" hexColour={'#' + this.state.blocks[2].hexColour}>
          <div className="page-home__screen-block--own-stuff__image-vertical-centerer-parent">
            <div className="page-home__screen-block--own-stuff__image-vertical-centerer-parent__image-vertical-centerer-child">
              <div className="page-home__screen-block--own-stuff__image-vertical-centerer-parent__image-vertical-centerer-child__image-layer">
                <svg className="page-home__screen-block--own-stuff__image-vertical-centerer-parent__image-vertical-centerer-child__image-layer__monument-award-1" role="presentation" dangerouslySetInnerHTML={{__html: MonumentAward1 }} />
                <svg className="page-home__screen-block--own-stuff__image-vertical-centerer-parent__image-vertical-centerer-child__image-layer__monument-award-2" role="presentation" dangerouslySetInnerHTML={{__html: MonumentAward2 }} />
                <svg className="page-home__screen-block--own-stuff__image-vertical-centerer-parent__image-vertical-centerer-child__image-layer__monument-award-3" role="presentation" dangerouslySetInnerHTML={{__html: MonumentAward3 }} />
                <MonumentDevice className="page-home__screen-block--own-stuff__image-vertical-centerer-parent__image-vertical-centerer-child__image-layer__monument-device" src="images/home/Homepage_Games_Ipad.png"/>
              </div>
            </div>
          </div>
          <div className="page-home__screen-block--own-stuff__text-block-vertical-centerer">
            <HomeTextBlock title="Award-winning own products and games" colour="nonBlack" childColour="nonBlack">
              <p>We understand that the best way to learn is by doing. We invest time and money to create our own products, testing ideas and concepts all the way from initial ideas to launch and beyond – learning and improving at every stage. With hugely successful, award-winning games like Monument Valley and innovative, problem-solving products like Wayfindr, we’ve proven that we can deliver. </p>
            </HomeTextBlock>
          </div>
        </ScreenBlock>
        <ScreenBlock ref="blockVenture" customClass="page-home__screen-block--ventures" hexColour={'#' + this.state.blocks[3].hexColour}>
          <div className="page-home__screen-block--ventures__image-vertical-centerer-parent">
            <div className="page-home__screen-block--ventures__image-vertical-centerer-parent__image-vertical-centerer-child">
              <div className="page-home__screen-block--ventures__image-vertical-centerer-parent__image-vertical-centerer-child__image-layer">
                <DiceScreenAnim className="page-home__screen-block--ventures__image-vertical-centerer-parent__image-vertical-centerer-child__image-layer__dice-screen-anim" src="images/home/Homepage_Dice_Motion.gif"/>
                <DiceDevice className="page-home__screen-block--ventures__image-vertical-centerer-parent__image-vertical-centerer-child__image-layer__dice-device" src="images/home/Homepage_Dice_Mask.png"/>
              </div>
            </div>
          </div>
          <div className="page-home__screen-block--ventures__text-block-vertical-centerer">
            <HomeTextBlock title="Launching new ventures" colour="white" childColour="white">
              <p>Collaboration is key. What excites us is working with people who know their industry inside out to create new and disruptive businesses. If we’re not starting a joint venture, then we’re finding and investing time and money in the most exciting ideas and startups.</p>
            </HomeTextBlock>
          </div>
        </ScreenBlock>
      </article>
    );
  }
};
