// templates/page-work.js
'use strict';

import React from 'react';

import WorkItem from 'components/work-item';

const work = [{
  id: 0,
  title: "Lush Digital",
  type: 'client',
  description: "Lorem ipsum dolor sit amet, consectetuer adipiscing elit, sed diam nonummy nibh euismod tincidunt.",
  image: '/images/whatwedo/projects/lush.png',
  link: '/work/lush',
  bgColour: 'rgba(165, 225, 255, 0.3)',
  primaryColour: 'rgba(97, 20, 204, 1)',
  secondaryColour: 'rgba(255, 191, 0, 1)'
}, {
  id: 1,
  title: "Monument Valley",
  type: 'ustwo games',
  description: "A story of forgiveness in which you guide Princess Ida through a world of impossible architecture.",
  image: '/images/whatwedo/projects/monument_valley.png',
  link: '/work/mv',
  bgColour: 'rgba(255, 195, 155, 0.3)',
  primaryColour: 'rgba(237, 0, 130, 1)',
  secondaryColour: 'rgba(22, 213, 217, 1)'
}, {
  id: 2,
  title: "Rewards by Harvey Nichols",
  type: 'client',
  description: "Lorem ipsum dolor sit amet, consectetuer adipiscing elit, sed diam nonummy nibh euismod tincidunt.",
  image: '/images/whatwedo/projects/harvey_nichols.png',
  link: '/work/hn',
  bgColour: 'rgba(234, 234, 234, 1)',
  primaryColour: 'rgba(100, 100, 100, 1)',
  secondaryColour: 'rgba(195, 130, 230, 1)'
}, {
  id: 3,
  title: "Google Watch Faces",
  type: 'client',
  description: "Lorem ipsum dolor sit amet, consectetuer adipiscing elit, sed diam nonummy nibh euismod tincidunt.",
  image: '/images/whatwedo/projects/google_wear.png',
  link: '/work/google',
  bgColour: 'rgba(165, 225, 255, 0.3)',
  primaryColour: 'rgba(97, 20, 204, 1)',
  secondaryColour: 'rgba(255, 191, 0, 1)'
}];

export default class PageWork extends React.Component {
  render() {
    return (
      <article className="page-work">
        <section className="page-work__hero" style={{backgroundImage: "url('/images/whatwedo/header/image_1.png')"}}>
          <h1 className="page-work__hero__title">Building and launching</h1>
          <img className="page-work__hero__image" src="/images/whatwedo/header/image_1.png" />
        </section>
        <section className="page-work__intro">
          <h2 className="page-work__intro__title">It's not what you say, it's what you do</h2>
          <hr className="page-work__intro__rule" />
          <p className="page-work__intro__para">We’re changing the experience of time-keeping for Google, driving for Jaguar and loyalty for Harvey Nics. We create and and bring our own successful products to market. We partner to launch new ventures that disrupt industries. Above all, our experience gives us the edge in building and launching digital products for others.</p>
        </section>
        <ul className="page-work__list">
          {work.map(workItem => <WorkItem key={workItem.id} className="page-work__list__item" data={workItem} />)}
        </ul>
      </article>
    );
  }
};
