// templates/page-what-we-do.js
'use strict';

import React from 'react';
import find from 'lodash/collection/find';
import get from 'lodash/object/get';

import WorkItem from 'components/work-item';
import DownChevron from 'elements/down-chevron';

const work = [{
  id: 0,
  title: "Lush Digital",
  type: 'client',
  description: "Lorem ipsum dolor sit amet, consectetuer adipiscing elit, sed diam nonummy nibh euismod tincidunt.",
  image: '/images/whatwedo/projects/lush.png',
  link: '/what-we-do/lush',
  bgColour: 'rgba(165, 225, 255, 0.3)',
  primaryColour: 'rgba(97, 20, 204, 1)',
  secondaryColour: 'rgba(255, 191, 0, 1)'
}, {
  id: 1,
  title: "Monument Valley",
  type: 'ustwo games',
  description: "A story of forgiveness in which you guide Princess Ida through a world of impossible architecture.",
  image: '/images/whatwedo/projects/monument_valley.png',
  link: '/what-we-do/mv',
  bgColour: 'rgba(255, 195, 155, 0.3)',
  primaryColour: 'rgba(237, 0, 130, 1)',
  secondaryColour: 'rgba(22, 213, 217, 1)'
}, {
  id: 2,
  title: "Rewards by Harvey Nichols",
  type: 'client',
  description: "Lorem ipsum dolor sit amet, consectetuer adipiscing elit, sed diam nonummy nibh euismod tincidunt.",
  image: '/images/whatwedo/projects/harvey_nichols.png',
  link: '/what-we-do/hn',
  bgColour: 'rgba(234, 234, 234, 1)',
  primaryColour: 'rgba(100, 100, 100, 1)',
  secondaryColour: 'rgba(195, 130, 230, 1)'
}, {
  id: 3,
  title: "Google Watch Faces",
  type: 'client',
  description: "Lorem ipsum dolor sit amet, consectetuer adipiscing elit, sed diam nonummy nibh euismod tincidunt.",
  image: '/images/whatwedo/projects/google_wear.png',
  link: '/what-we-do/google',
  bgColour: 'rgba(165, 225, 255, 0.3)',
  primaryColour: 'rgba(97, 20, 204, 1)',
  secondaryColour: 'rgba(255, 191, 0, 1)'
}];

export default class PageWork extends React.Component {
  render() {
    const pageData = this.props.page;
    return (
      <article className="page-work">
        <section className="page-work__hero" style={{backgroundImage: "url('/images/whatwedo/header/image_1.png')"}}>
          <h1 className="page-work__hero__title">{get(pageData, 'page_builder.0.attr.heading.value')}</h1>
          <img className="page-work__hero__image" src="/images/whatwedo/header/image_1.png" />
          <DownChevron customClass="page-work__hero__down-chevron" ref="downChevron" />
        </section>
        <section className="page-work__intro">
          <h2 className="page-work__intro__title">{get(pageData, 'page_builder.1.attr.heading.value')}</h2>
          <hr className="page-work__intro__rule" />
          <div className="page-work__intro__para" dangerouslySetInnerHTML={{__html: get(pageData, 'page_builder.1.attr.body.value')}} />
        </section>
        <ul className="page-work__list">
          {get(pageData, 'page_builder.2.attr.case_studies.value', []).map(caseStudyName => {
            console.log(caseStudyName);
            const caseStudyData = find(get(pageData, '_embedded.ustwo:case_studies', []), 'slug', caseStudyName);
            return <WorkItem key={caseStudyName} className="page-work__list__item" data={caseStudyData} />;
          })}
        </ul>
      </article>
    );
  }
  componentDidMount() {
    setTimeout(() => {
      this.refs.downChevron.resetAnim();
      this.refs.downChevron.anim();
    }, 500);
  }
}
