// components/blog-post-list-item.js
import React from 'react';
import classnames from 'classnames';

import Actions from 'flux/actions';

let dataPost = {
  title: "Humanizing Wearables",
  imageURL: "/images/blog/temp-image.jpg",
  category: "Design",
  author: "Mills",
  date: "13 March 2015",
  summary: "Lorem ipsum dolor sit amet consectetuer adipiscing elit, sed diam nonummy nibh euismod tincidunt ui laoreet dolore magna aliquam erat"
}

export default class BlogPostListItem extends React.Component {
  render() {
    const classes = classnames('blog-post-list-item', dataPost.category.toLowerCase(), {
      featured: this.props.featured
    });
    return (
      <article className={classes}>
        <div className="image" style={{backgroundImage: `url(${dataPost.imageURL})`}}>
          <img src={dataPost.imageURL} />
        </div>
        <div className="content">
          <div className="category">{dataPost.category}</div>
          <h2>{dataPost.title}</h2>
          <div className="credits">
            <span className="by">By {dataPost.author}</span> - {dataPost.date}
          </div>
          <p>{dataPost.summary}</p>
          <div className="tail">
            <a href="/">Read more</a>
            <div className="social">
              <div className="twitter">120</div>
              <div className="comments">43</div>
            </div>
          </div>
        </div>
      </article>
    );
  }
}
