// components/contact-tray.js
import React from 'react';

import Actions from 'flux/actions';
import ContactDetail from 'components/contact-detail';
import CloseButton from 'elements/close-button';
import {onClickContent} from 'modules/modal';

const contactData = [{
  title: "Work with us",
  type: "work",
  desc: "Have a project or an idea you’d like to collaborate with ustwo on? Let’s talk.",
  methods: [{
    text: "work@ustwo.com",
    url: "work@ustwo.com",
    type: "email"
  }]
}, {
  title: "Looking for your next job?",
  type: "job",
  desc: "We’re always on the lookout  for the most talented designers, engineers and business people. If that’s you, why didn’t you say earlier?",
  methods: [{
    text: "Check out our jobs listings",
    url: "/jobs",
    type: "link"
  }]
}, {
  title: "Press enquiries",
  type: "press",
  desc: "We love a good story. For all press opportunities, please reach out to our team.",
  methods: [{
    text: "press@ustwo.com",
    url: "press@ustwo.com",
    type: "email"
  }]
}, {
  title: "General enquiries",
  type: "general",
  desc: "Questions? Comments? Smart remarks? Shoot us an email or give us a ring.",
  methods: [{
    text: "hello@ustwo.com",
    url: "hello@ustwo.com",
    type: "email"
  }, {
    text: "+44 (0) 207 613 0433",
    url: "+4402076130433",
    type: "tel"
  }]
}];

export default class ContactTray extends React.Component {
  render() {
    return (
      <div className="contact-tray" onClick={onClickContent}>
        <CloseButton onClose={this.onClickClose} className="contact-tray__close" autoAnim={500} />
        <article className="contact-tray__content">
          {contactData.map(contactDetail => <ContactDetail detail={contactDetail}/>)}
        </article>
      </div>
    );
  }
  onClickClose() {
    Actions.closeModal();
  }
}
