// components/contact-tray.js
import React from 'react';

import Actions from 'flux/actions';
import ContactDetail from 'components/contact-detail';
import CloseButton from 'elements/close-button';

const contactData = [{
  title: "Need our skills on a project?",
  description: "If you’d like to talk about working with us, drop us an email and we’ll get the ball rolling.",
  links: [{
    text: "work@ustwo.com",
    url: "work@ustwo.com",
    type: "email"
  }]
}, {
  title: "Looking for your next job?",
  description: "We’re always looking for talented people. If that’s you, why didn’t you say earlier?",
  links: [{
    text: "Check out our jobs listings",
    url: "/jobs",
    type: "link"
  }]
}, {
  title: "Press enquiries",
  description: "For all things press, send us a mail and we can talk.",
  links: [{
    text: "press@ustwo.com",
    url: "press@ustwo.com",
    type: "email"
  }]
}, {
  title: "General enquiries",
  description: "What do you want to know? Send us a mail or try us on the phone.",
  links: [{
    text: "hello@ustwo.com",
    url: "hello@ustwo.com",
    type: "email"
  }, {
    text: "+44 (0) 207 613 0433",
    url: "+4402076130433",
    type: "tel"
  }]
}];

export default class ContactTray extends React.Component {
  render() {
    return (
      <div className="contact-tray">
        <CloseButton ref="close" onClose={this.onClickClose} className="contact-tray__close" />
        <article className="contact-tray__content">
          {contactData.map(this.renderContactDetail)}
        </article>
      </div>
    );
  }
  componentDidMount() {
    setTimeout(() => {
      this.refs.close.anim();
    }, 500);
  }
  renderContactDetail(contactDetail) {
    return <ContactDetail detail={contactDetail}/>;
  }
  onClickClose() {
    Actions.toggleContacts();
  }
};
