// components/studio-contact.js
'use strict';

import React from 'react';
import classnames from 'classnames';

export default class StudioContact extends React.Component {
  constructor(props) {
    super(props);
    this.state = {
      open: false
    };
  }
  onClickTitle = () => {
    this.setState({
      open: !this.state.open
    });
  }
  render() {
    const studio = this.props.studio;
    const style = this.state.open ? {
      backgroundColor: studio.color,
      borderBottomColor: studio.color
    } : {};
    const mapurl = `https://maps.google.com/maps?z=12&t=m&q=loc:${studio.location.lat}+${studio.location.long}`;
    return (
      <li className={classnames('studio', {
          open: this.state.open
        })} style={style}>
        <h1 className="studio-title" onClick={this.onClickTitle}>{studio.name}</h1>
        <div className="studio-details">
          <div className="vcard">
            <div className="adr">
              <p className="street-address">{studio.address[0]}</p>
              <p className="place">{studio.address[1]}</p>
              <p className="country-name">{studio.address[2]}</p>
            </div>
            <a className="url map" href={mapurl}>Google Maps</a>
          </div>
        </div>
      </li>
    );
  }
};
