// _lib/fetcher.js
import { polyfill } from 'es6-promise';
import Fetch from 'Fetch';

import pick from 'lodash/object/pick';

(typeof window !== 'undefined') && (window.ajaxPending = false);
let ajaxes = {};

// TODO: handle different environments
let defaultConfig = {
  baseurl: 'http://ustwo-staging.aws.hmn.md/wp-json/'
}

function fetcher (config) {
  const req = Fetch(config.url, Object.assign({
      method: 'get',
      credentials: 'include'
    }, pick(config, ['method', 'body'])))
    .then((response) => {
      remove(config.url);
      if (response.status >= 400) {
        if(config.failure) {
          config.failure(response);
        } else {
          throw new Error("Bad response from server");
        }
      }
      return response.json();
    })
    .then(config.success)
    .catch(config.failure);
  add(config.url, req);
}

function add (url, req) {
  ajaxes[url] = req;
  window.ajaxPending = true;
}

function remove (url) {
  delete ajaxes[url];
  window.ajaxPending = !!Object.keys(ajaxes).length;
}

export default fetcher;
