// flux/routes.js
import Actions from 'flux/actions';

const handlers = {
  home() {
    Actions.goTo('home');
  },
  whatWeDo() {
    Actions.goTo('what-we-do', [{
      namespace: 'wp/v2/',
      type: 'pages',
      id: '66?_embed'
    }]);
  },
  caseStudy(cid) {
    Actions.goTo('what-we-do/case-study', [{
      namespace: 'ustwo/v1/',
      type: 'case-studies',
      id: cid
    }]);
  },
  blog() {
    Actions.goTo('blog', [{
      namespace: 'wp/v2/',
      type: 'pages',
      id: '?name=blog&_embed=true',
      get: data => data[0]
    }]);
  },
  post(pid) {
    Actions.goTo('blog/post', [{
      namespace: 'wp/v2/',
      type: 'posts',
      id: `?name=${pid}&_embed=true`,
      get: data => data[0]
    }]);
  },
  legal() {
    Actions.goTo('legal', [{
      namespace: 'wp/v2/',
      type: 'pages',
      id: '?name=legal&_embed=true',
      get: data => data[0]
    }]);
  },
  privacy() {
    Actions.goTo('privacy', [{
      namespace: 'wp/v2/',
      type: 'pages',
      id: '?name=privacy&_embed=true',
      get: data => data[0]
    }]);
  }
};

/**
  Example route definitions
  {
    '/': function () {},
    '/projects': function () {},
    '/projects/:pid': function (pid) {}
  }
*/
const routes = {
  home: {
    pattern: '/',
    handler: handlers.home
  },
  work: {
    pattern: '/what-we-do',
    handler: handlers.whatWeDo
  },
  caseStudy: {
    pattern: '/what-we-do/:cid',
    handler: handlers.caseStudy
  },
  blog: {
    pattern: '/blog',
    handler: handlers.blog
  },
  post: {
    pattern: '/blog/:pid',
    handler: handlers.post
  },
  legal: {
    pattern: '/legal',
    handler: handlers.legal
  },
  privacy: {
    pattern: '/privacy',
    handler: handlers.privacy
  }
};

export default routes;
