// flux/store.js
import {EventEmitter} from 'events';
import assign from 'lodash/object/assign';
import clone from 'lodash/lang/cloneDeep';
import capitalize from 'lodash/string/capitalize';
import 'whatwg-fetch';

import Nulls from 'flux/nulls';

let _data = {};

const _state = {
  page: Nulls.page,
  modal: Nulls.modal,
  takeover: Nulls.takeover,
  colours: Nulls.colours,
  pages: Nulls.pages
};

const Store = assign({}, EventEmitter.prototype, {
  getData() {
    return _state;
  },
  emitChange() {
    Store.emit('change');
  },
  addChangeListener(callback) {
    Store.on('change', callback);
  },
  removeChangeListener(callback) {
    Store.removeListener('change', callback);
  }
});

window.Store = Store;

// This is currently broken in IE11: https://github.com/github/fetch/issues/114
fetch('/data/gulpdata.json')
  .then((response) => {
    return response.json();
  }).then((json) => {
    _data = json;
    Object.assign(_state, json);
    if(window.localStorage.getItem(_state.takeover.id)) {
      _state.takeover = Nulls.takeover;
    }
    Store.emitChange();
  }).catch(function(ex) {
    console.warn('App data JSON parsing failed:', ex);
  });

export default {
  init: Store.init,
  getData: Store.getData,
  emitChange: Store.emitChange,
  addChangeListener: Store.addChangeListener,
  removeChangeListener: Store.removeChangeListener,

  setPage(newPage) {
    _state.page = newPage;
  },
  toggleContacts() {
    _state.modal = _state.modal !== 'contacts' ? 'contacts' : false;
  },
  closeTakeover() {
    window.localStorage.setItem(_state.takeover.id, true);
    _state.takeover = Nulls.takeover;
  },
  closeModal() {
    _state.modal = false;
  }
};
